### DISCUSSION TITLE
Monopoly Clause improvements

### DISCUSSION THREAD

**Discussion by samdowd** [kinfoundation/kin-rfcs#132] [_ Archive - KRE 3.1 Discussion]
I think the improvement to the monopoly clause is a step in the right direction. Giving each app some seed Kin as part of the KRE is good for all apps in the long term. But I'd further argue that incentivizing apps at every level is good for all apps in the long term. This new improvement only helps very small apps. And the standing monopoly clause is pretty limited. It made more sense when there were fewer apps in the ecosystem, but as more apps join, having a hardcoded monopoly value of 1 or 2 makes it less and less meaningful. I believe a stronger monopoly clause could help with the valley in KRE rewards between very small and very large apps. And I believe that removing say 20MM Kin from the top 3 apps and distributing it among the smaller apps would have a long term positive effect on the value of the top 3 apps' rewards on top of helping everyone else.

The marginal value of 1 Kin to an app making 10MM is much lower than to an app making 100,000. In addition, the marginal value to the ecosystem of an app going from 100 to 101 users is higher than an app going from 100,000 to 100,0001. Given that these two values are so well aligned, the KRE could be adjusted to distribute the rewards accordingly.

My rough proposal is to make a slightly more robust change to the monopoly clause. In place of the `(2,999 * Payout_i / 3,000) + (MAX(Payout) * (1 / 3,000))` formula proposed, I propose something along the lines of `Payout_i + 5*ln(Payout_i)`. Without access to the underlying data, I don't know what the best constants would be to apply the right amount of smoothness to this, but the basic concept is very similar to the current proposal. Use the base payout rate primarily, but with an added component to ease the curve a little for smaller values. This is what my curve looks like, x symbolizing pre-adjusted payout, and y symbolizing adjusted payout:
<img width="468" alt="image" src="https://user-images.githubusercontent.com/8724612/153772239-4f55a660-c533-4c19-82b4-67416f3fa618.png">

As you can see, the result is that the beginning of the curve (the first 100 users or 2000 Kin median spend, for example) increases more quickly, showing rewards increasing more quickly. Then the curve slows down until it's barely different from a straight line as the `ln(Payout_i)` portion of the equation becomes less and less meaningful at larger numbers. Obviously the specific values here are just demonstrative, but that shape of curve is the key to a KRE that rewards everyone big and small, I think.

**Comment by skyf0xx** [comment]
@samdowd An interesting proposal in addition to this one: https://github.com/kinecosystem/rewards-engine/discussions/130
I will do some math on it and see how it plays out, for further discussion. Note though that our update to the clause didn't the negate the previous one but added to it.

**Reply by jongarsman1076** [reply]
This is great Sam!  I felt the same way but not quite bright enough to share a solution.  Thank you!

**Reply by samdowd** [reply]
@hitwill Yeah I understand. I'm proposing that this solution would replace the 3.1 change but leave the 3.0 clause in place. Just like the current proposal does, just with a different equation.

**Comment by Freedom-kin** [comment]
Why do we want developers to compete with each other for KRE?

App X can do very well, but than a very large app Y joins the system and all of the sudden app X get a lot less KRE.

App Y can get up to 66.67% of the total KRE.
Even worse if a second very large app joins,
The 2 can get up to 90%.

All other apps are left with dividing just 10%
Not nice if there are 20 apps, but a lot worse if there will be 50 apps in the near future.

If you just reward every app with 30% reward on their net SPENDS (net turnover) with a cap of 50MM, than no one has to compete. One or two very big apps don't have any effect on other apps.
You don't need further monopoly clauses. 

It's real simple, no other factors for a KRE score and the amount of available KRE grows (or shrinks) with the number of apps and how well they perform.
